### PR TITLE
[patch] MASCORE-2054: DRO/UDS storageclass fix

### PR DESCRIPTION
**Pull request by yuvraj-vansure**
[MASCORE-2054](https://jsw.ibm.com/browse/MASCORE-2054): Added storageclass check in dro and uds role, to use default annotated storageclass, if value is not provided by user. 
[MASCORE-2055](https://jsw.ibm.com/browse/MASCORE-2055): Added migration_status check for uninstall UDS

### CODE REVIEW

**Review by lokesh-sreedhara — DISMISSED** [review]

File: ibm/mas_devops/roles/dro/tasks/main.yml [hunk 1]
```diff
@@ -29,7 +29,14 @@
...
   when:
-    - dro_migration != "" and dro_migration | lower == "true"
+    - dro_migration != ""
+    - dro_migration | lower == "true"
+  block:
+    - set_fact:
+        migration_status: "installing-dro"
+
+    - name: install dro
+      include_tasks: "tasks/install-dro/main.yml"
```
Comment (whitfiea) [inline ibm/mas_devops/roles/dro/tasks/main.yml:39]: We are calling install-dro/main.yml twice?

File: ibm/mas_devops/roles/uds/tasks/install/determine-storage-classes.yml [hunk 2]
```diff
@@ -26,6 +26,11 @@
   set_fact:
     uds_storage_class: "{{ lookup_storageclasses | ibm.mas_devops.defaultStorageClass(default_storage_classes_rwo) }}"
 
+- name: Default annotated storage if not found in default list
+  when: uds_storage_class is not defined or uds_storage_class == ""
+  set_fact:
+    uds_storage_class: "{{ lookup_storageclasses | ibm.mas_devops.getDefaultStorageClass() }}"
```
Comment (whitfiea) [inline ibm/mas_devops/roles/uds/tasks/install/determine-storage-classes.yml:32]: Does UDS work with any access mode?

File: ibm/mas_devops/roles/dro/tasks/install-dro/determine-storage-classes.yml [hunk 3]
```diff
@@ -21,10 +21,18 @@
...
+  block:
+    - name: Default Storage for ROKS if not set by user
+      when: dro_storage_class is not defined or dro_storage_class == ""
+      set_fact:
+        dro_storage_class: "{{ lookup_storageclasses | ibm.mas_devops.defaultStorageClass(default_storage_classes_rwo) }}"
+
+    - name: Default annotated storage if not found in default list
+      when: dro_storage_class is not defined or dro_storage_class == ""
+      set_fact:
+        dro_storage_class: "{{ lookup_storageclasses | ibm.mas_devops.getDefaultStorageClass() }}"
```
Comment (whitfiea) [inline ibm/mas_devops/roles/dro/tasks/install-dro/determine-storage-classes.yml:35]: Does DRO work with any access mode?

**Review by whitfiea — CHANGES_REQUESTED** [review]

File: ibm/mas_devops/roles/dro/tasks/main.yml (hunk 1 above)
Comment (yuvraj-vansure) [reply ibm/mas_devops/roles/dro/tasks/main.yml:39]: `install-dro` is called only once depending on the dro_migration value.

File: ibm/mas_devops/roles/uds/tasks/install/determine-storage-classes.yml (hunk 2 above)
Comment (yuvraj-vansure) [reply ibm/mas_devops/roles/uds/tasks/install/determine-storage-classes.yml:32]: UDS requires rwo storageclass. Just added this condition to check and use the storgaclass with default annotation if default_storage_classes_rwo list returns no result.

File: ibm/mas_devops/roles/dro/tasks/main.yml (hunk 1 above)
Comment (whitfiea) [reply ibm/mas_devops/roles/dro/tasks/main.yml:39]: Are you sure? 
image

File: ibm/mas_devops/roles/uds/tasks/install/determine-storage-classes.yml (hunk 2 above)
Comment (whitfiea) [reply ibm/mas_devops/roles/uds/tasks/install/determine-storage-classes.yml:32]: I guess the problem is what happens if the default storage class is rwx

**Review by whitfiea — CHANGES_REQUESTED** [review]

File: ibm/mas_devops/roles/dro/tasks/main.yml (hunk 1 above)
Comment (yuvraj-vansure) [reply ibm/mas_devops/roles/dro/tasks/main.yml]: ohh. my bad, removed that block as it is not required.

File: ibm/mas_devops/roles/uds/tasks/install/determine-storage-classes.yml (hunk 2 above)
Comment (yuvraj-vansure) [reply ibm/mas_devops/roles/uds/tasks/install/determine-storage-classes.yml:32]: I think, If the default storage class is only rwx then, pvc creation might fail.

File: ibm/mas_devops/roles/dro/tasks/install-dro/determine-storage-classes.yml (hunk 3 above)
Comment (yuvraj-vansure) [reply ibm/mas_devops/roles/dro/tasks/install-dro/determine-storage-classes.yml:35]: confirmed from Daniel Cleyrat that DRO uses

 RWO is the mode for the default PVC, they have not tested any other modes.

**Review by yuvraj-vansure — COMMENTED** [review]
> I guess the problem is what happens if the default storage class is rwx

Selecting the default storage class which may not be WXO can cause issues. Making sure the storage_class variable is not empty by adding another potential issue does not make sense.
Closing this PR as these changes are not valid.

**Pull request closed without merging**